File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/Attributes.kt
package com.nouprax.markdown.core

public data class Record(
    public val name: String,
    public val value: String,
)

/** Ordered immutable values, including every duplicate. */
public class Attributes(
    classes: kotlin.collections.List<String>,
    records: kotlin.collections.List<Record>,
) {
    public val classes: kotlin.collections.List<String> = classes.immutableMap { it }
    public val records: kotlin.collections.List<Record> = records.immutableMap { it }

    public companion object {
        public val empty: Attributes = Attributes(emptyList(), emptyList())
    }
}

File: packages/es-markdown-core/src/markup/attributes.ts
export interface Record {
    readonly name: string;
    readonly value: string;
}
export interface Attributes {
    readonly classes: readonly string[];
    readonly records: readonly Record[];
}
export const Attributes: { readonly empty: Attributes } = Object.freeze({
    empty: Object.freeze({ classes: Object.freeze([]), records: Object.freeze([]) })
});
